src/Compilers/CSharp/Test/Semantic/Semantics/UserDefinedConversionTests.cs

```diff
@@ -1647,5 +1647,266 @@ private static C<object> M2()
                 //         return (C<object>) M1();
                 Diagnostic(ErrorCode.ERR_NoExplicitConv, "(C<object>) M1()").WithArguments("void", "C<object>").WithLocation(9, 16));
         }
+
+        [Fact, WorkItem(56646, "https://github.com/dotnet/roslyn/issues/56646")]
+        public void LiftedConversion_InvalidTypeArgument01()
+        {
+            var code = @"
+int? i = null;
+C c = i;
+
+class C
+{
+    public static implicit operator C(long l) => throw null;
+}
+
+namespace System
+{
+    public class Object {}
+    public class String {}
+    public class Exception {}
+    public class ValueType : Object {}
+    public struct Void {}
+    public struct Int32 {}
+    public struct Nullable<T> where T : struct {}
+    public ref struct Int64 {}
+}
+";
+
+            var comp = CreateEmptyCompilation(code);
+            comp.VerifyDiagnostics(
+                // (3,7): error CS0266: Cannot implicitly convert type 'int?' to 'C'. An explicit conversion exists (are you missing a cast?)
+                // C c = i;
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "i").WithArguments("int?", "C").WithLocation(3, 7)
+            );
+        }
+
+        [Fact, WorkItem(56646, "https://github.com/dotnet/roslyn/issues/56646")]
+        public void LiftedConversion_InvalidTypeArgument02()
+        {
+            var code = @"
+C c = null;
+M(c);
+
+void M(long? i) => throw null;
+
+class C
+{
+    public static implicit operator int(C c) => throw null;
+}
+
+namespace System
+{
+    public class Object {}
+    public class String {}
+    public class Exception {}
+    public class ValueType : Object {}
+    public struct Void {}
+    public ref struct Int32 {}
+    public struct Nullable<T> where T : struct { public Nullable(T value) {} }
+    public struct Int64 {}
+    public class Attribute {}
+}
+";
+
+            var comp = CreateEmptyCompilation(code);
+            var verifier = CompileAndVerify(comp);
```

Comment:
>CompileAndVerify [](http://example.com/codeflow?start=27&length=16)

It looks like verification should be disabled on desktop for test using fake special types.